crates/ethcore/src/engines/authority_round/mod.rs

```diff
@@ -113,7 +113,24 @@ impl From<ethjson::spec::AuthorityRoundParams> for AuthorityRoundParams {
             validate_score_transition: p.validate_score_transition.map_or(0, Into::into),
             validate_step_transition: p.validate_step_transition.map_or(0, Into::into),
             immediate_transitions: p.immediate_transitions.unwrap_or(false),
-            block_reward: p.block_reward.map_or_else(Default::default, Into::into),
+            block_reward: p.block_reward.map_or_else(
+                || {
+                    let mut ret = BTreeMap::new();
+                    ret.insert(0, U256::zero());
+                    ret
+                },
+                |reward| match reward {
+                    ethjson::spec::BlockReward::Single(reward) => {
+                        let mut ret = BTreeMap::new();
+                        ret.insert(0, reward.into());
+                        ret
+                    }
+                    ethjson::spec::BlockReward::Multi(multi) => multi
```

Comment:
I think that we have to check here that there is the block zero defined, instead of crashing later with `Current block's reward is not found; this indicates a chain config error; qed`. Ideally implement `try_from` instead `from`